INFO 2024-02-19 21:07:00,265 autoreload 484 20736 Watching for file changes with StatReloader
WARNING 2024-02-19 21:07:02,209 log 484 2572 Not Found: /
WARNING 2024-02-19 21:07:02,210 basehttp 484 2572 "GET / HTTP/1.1" 404 2170
INFO 2024-02-19 21:07:16,743 basehttp 484 2572 "GET /shop HTTP/1.1" 301 0
INFO 2024-02-19 21:07:16,745 views 484 5932 Info page object accessed
INFO 2024-02-19 21:07:16,752 basehttp 484 5932 "GET /shop/ HTTP/1.1" 200 1693
INFO 2024-02-19 21:07:16,769 basehttp 484 22924 "GET /static/css/base.css HTTP/1.1" 304 0
INFO 2024-02-19 21:07:16,811 basehttp 484 5932 "GET /static/img/logo.png HTTP/1.1" 200 7903
WARNING 2024-02-19 21:07:16,811 basehttp 484 22924 "GET /static/main/img/Logo_2.png HTTP/1.1" 404 1816
INFO 2024-02-19 21:07:19,421 views 484 5932 Info page object accessed
INFO 2024-02-19 21:07:19,423 basehttp 484 5932 "GET /shop/about_us/ HTTP/1.1" 200 3192
WARNING 2024-02-19 21:07:19,481 basehttp 484 5932 "GET /static/main/img/Logo_2.png HTTP/1.1" 404 1816
INFO 2024-02-19 21:07:20,315 views 484 22180 Info page object accessed
INFO 2024-02-19 21:07:20,315 basehttp 484 22180 "GET /shop/ HTTP/1.1" 200 1693
WARNING 2024-02-19 21:07:20,350 basehttp 484 22180 "GET /static/main/img/Logo_2.png HTTP/1.1" 404 1816
INFO 2024-02-24 20:36:55,203 autoreload 1380 19484 Watching for file changes with StatReloader
WARNING 2024-02-24 20:36:57,717 log 1380 9608 Not Found: /
WARNING 2024-02-24 20:36:57,717 basehttp 1380 9608 "GET / HTTP/1.1" 404 2170
WARNING 2024-02-24 20:36:57,858 log 1380 9608 Not Found: /favicon.ico
WARNING 2024-02-24 20:36:57,859 basehttp 1380 9608 "GET /favicon.ico HTTP/1.1" 404 2221
INFO 2024-02-24 21:20:48,861 autoreload 19952 7548 Watching for file changes with StatReloader
ERROR 2024-02-24 21:20:51,469 log 19952 18304 Internal Server Error: /
Traceback (most recent call last):
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py", line 5, in order
    return render(request,'myapp_2/')
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\loader.py", line 15, in get_template
    return engine.get_template(template_name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\backends\django.py", line 33, in get_template
    return Template(self.engine.get_template(template_name), self)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\engine.py", line 177, in get_template
    template, origin = self.find_template(template_name)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\engine.py", line 159, in find_template
    template = loader.get_template(name, skip=skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\loaders\cached.py", line 57, in get_template
    template = super().get_template(template_name, skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\loaders\base.py", line 23, in get_template
    contents = self.get_contents(origin)
               ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\loaders\cached.py", line 26, in get_contents
    return origin.loader.get_contents(origin)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\loaders\filesystem.py", line 22, in get_contents
    with open(origin.name, encoding=self.engine.file_charset) as fp:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
PermissionError: [Errno 13] Permission denied: 'C:\\Users\\Никита\\Desktop\\DJANGO\\ShopProject\\Shop_project_HW\\shop_project\\myapp_2\\templates\\myapp_2'
ERROR 2024-02-24 21:20:51,474 basehttp 19952 18304 "GET / HTTP/1.1" 500 111024
INFO 2024-02-24 21:22:33,243 autoreload 19952 7548 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py changed, reloading.
INFO 2024-02-24 21:22:33,674 autoreload 20784 18128 Watching for file changes with StatReloader
INFO 2024-02-24 21:22:48,096 autoreload 20784 18128 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py changed, reloading.
INFO 2024-02-24 21:22:48,501 autoreload 20244 4240 Watching for file changes with StatReloader
INFO 2024-02-24 21:22:53,782 autoreload 20244 4240 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py changed, reloading.
INFO 2024-02-24 21:22:54,240 autoreload 6632 2664 Watching for file changes with StatReloader
ERROR 2024-02-24 21:22:59,633 log 6632 12576 Internal Server Error: /
Traceback (most recent call last):
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py", line 5, in order
    return render(request, "myapp_2/order_client.html")
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\loader.py", line 15, in get_template
    return engine.get_template(template_name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\backends\django.py", line 33, in get_template
    return Template(self.engine.get_template(template_name), self)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\engine.py", line 177, in get_template
    template, origin = self.find_template(template_name)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\engine.py", line 159, in find_template
    template = loader.get_template(name, skip=skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\loaders\cached.py", line 57, in get_template
    template = super().get_template(template_name, skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\loaders\base.py", line 28, in get_template
    return Template(
           ^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 154, in __init__
    self.nodelist = self.compile_nodelist()
                    ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 196, in compile_nodelist
    return parser.parse()
           ^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 511, in parse
    self.extend_nodelist(nodelist, compiled_result, token)
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 528, in extend_nodelist
    raise self.error(
django.template.exceptions.TemplateSyntaxError: <ExtendsNode: extends 'myapp_2/main.html'> must be the first tag in the template.
ERROR 2024-02-24 21:22:59,637 basehttp 6632 12576 "GET / HTTP/1.1" 500 125650
INFO 2024-02-24 21:29:00,159 basehttp 6632 10664 "GET / HTTP/1.1" 200 1707
INFO 2024-02-24 21:29:00,235 basehttp 6632 17348 "GET /static/img/logo.png HTTP/1.1" 200 7903
INFO 2024-02-24 21:29:00,235 basehttp 6632 10664 "GET /static/css/base.css HTTP/1.1" 200 9947
WARNING 2024-02-24 21:29:00,306 basehttp 6632 10664 "GET /static/main/img/Logo_2.png HTTP/1.1" 404 1816
WARNING 2024-02-24 21:29:00,669 basehttp 6632 17348 "GET /static/main/img/Logo_2.png HTTP/1.1" 404 1816
INFO 2024-02-24 21:30:35,485 basehttp 6632 19960 "GET / HTTP/1.1" 200 1707
WARNING 2024-02-24 21:30:35,542 basehttp 6632 19960 "GET /static/main/img/Logo_2.png HTTP/1.1" 404 1816
INFO 2024-02-24 21:30:39,443 basehttp 6632 19544 "GET / HTTP/1.1" 200 1707
WARNING 2024-02-24 21:30:39,482 basehttp 6632 19544 "GET /static/main/img/Logo_2.png HTTP/1.1" 404 1816
WARNING 2024-02-24 21:31:33,783 log 6632 1920 Not Found: /order_client
WARNING 2024-02-24 21:31:33,784 basehttp 6632 1920 "GET /order_client HTTP/1.1" 404 2327
INFO 2024-02-24 21:31:42,979 basehttp 6632 1920 "GET / HTTP/1.1" 200 1707
WARNING 2024-02-24 21:31:43,024 basehttp 6632 1920 "GET /static/main/img/Logo_2.png HTTP/1.1" 404 1816
INFO 2024-02-24 21:33:01,372 basehttp 6632 2936 "GET / HTTP/1.1" 200 1767
WARNING 2024-02-24 21:33:01,424 basehttp 6632 2936 "GET /static/main/img/Logo_2.png HTTP/1.1" 404 1816
INFO 2024-02-24 22:25:26,330 autoreload 6632 2664 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py changed, reloading.
INFO 2024-02-24 22:25:27,158 autoreload 4780 7440 Watching for file changes with StatReloader
INFO 2024-02-24 22:26:25,255 autoreload 4780 7440 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py changed, reloading.
INFO 2024-02-24 22:26:25,944 autoreload 12764 8892 Watching for file changes with StatReloader
INFO 2024-02-24 22:26:55,685 autoreload 12764 8892 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py changed, reloading.
INFO 2024-02-24 22:26:56,260 autoreload 21216 6404 Watching for file changes with StatReloader
INFO 2024-02-24 22:27:12,576 autoreload 21216 6404 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py changed, reloading.
INFO 2024-02-24 22:27:12,969 autoreload 19920 20896 Watching for file changes with StatReloader
INFO 2024-02-24 22:27:29,437 autoreload 19920 20896 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py changed, reloading.
INFO 2024-02-24 22:27:29,862 autoreload 2928 16596 Watching for file changes with StatReloader
INFO 2024-02-24 22:27:43,262 autoreload 2928 16596 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py changed, reloading.
INFO 2024-02-24 22:27:43,663 autoreload 11724 7440 Watching for file changes with StatReloader
INFO 2024-02-24 22:28:09,237 autoreload 11724 7440 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py changed, reloading.
INFO 2024-02-24 22:28:09,790 autoreload 17768 11248 Watching for file changes with StatReloader
INFO 2024-02-24 22:30:15,185 autoreload 17768 11248 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py changed, reloading.
INFO 2024-02-24 22:30:15,950 autoreload 21440 1416 Watching for file changes with StatReloader
INFO 2024-02-24 22:31:11,183 autoreload 21440 1416 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py changed, reloading.
INFO 2024-02-24 22:31:11,614 autoreload 1796 20008 Watching for file changes with StatReloader
INFO 2024-02-24 22:31:43,821 basehttp 1796 5844 "GET / HTTP/1.1" 200 1918
INFO 2024-02-24 22:31:43,832 basehttp 1796 5844 "GET /static/css/base.css HTTP/1.1" 304 0
INFO 2024-02-24 22:31:43,832 basehttp 1796 10492 "GET /static/img/logo.png HTTP/1.1" 304 0
WARNING 2024-02-24 22:31:43,919 basehttp 1796 10492 "GET /static/main/img/Logo_2.png HTTP/1.1" 404 1816
INFO 2024-02-24 22:36:18,990 autoreload 5036 18500 Watching for file changes with StatReloader
INFO 2024-02-24 22:36:20,722 basehttp 5036 6884 "GET / HTTP/1.1" 200 1918
INFO 2024-02-24 22:36:20,784 basehttp 5036 6884 "GET /static/img/logo.png HTTP/1.1" 200 7903
WARNING 2024-02-24 22:36:20,858 basehttp 5036 6884 "GET /static/main/img/Logo_2.png HTTP/1.1" 404 1816
INFO 2024-02-24 22:36:22,726 basehttp 5036 2700 "GET / HTTP/1.1" 200 1918
WARNING 2024-02-24 22:36:22,764 basehttp 5036 2700 "GET /static/main/img/Logo_2.png HTTP/1.1" 404 1816
INFO 2024-02-24 22:37:00,619 basehttp 5036 9028 "GET / HTTP/1.1" 200 1918
WARNING 2024-02-24 22:37:00,664 basehttp 5036 9028 "GET /static/main/img/Logo_2.png HTTP/1.1" 404 1816
INFO 2024-02-24 22:38:43,858 autoreload 5036 18500 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py changed, reloading.
INFO 2024-02-24 22:38:44,285 autoreload 18492 19840 Watching for file changes with StatReloader
INFO 2024-02-24 22:40:32,938 autoreload 19808 3640 Watching for file changes with StatReloader
INFO 2024-02-24 22:41:32,849 autoreload 19808 3640 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\shop_project\urls.py changed, reloading.
INFO 2024-02-24 22:41:33,397 autoreload 10796 8788 Watching for file changes with StatReloader
INFO 2024-02-24 22:41:33,773 basehttp 10796 18516 "GET / HTTP/1.1" 200 2076
WARNING 2024-02-24 22:41:33,854 basehttp 10796 18516 "GET /static/main/img/Logo_2.png HTTP/1.1" 404 1816
WARNING 2024-02-24 22:42:17,137 basehttp 10796 9524 "GET /static/main/img/Logo_2.png HTTP/1.1" 404 1816
WARNING 2024-02-24 22:42:24,628 log 10796 2768 Not Found: /order_client/<int:client_id>/
WARNING 2024-02-24 22:42:24,630 basehttp 10796 2768 "GET /order_client/%3Cint:client_id%3E/ HTTP/1.1" 404 2394
WARNING 2024-02-24 22:42:43,079 log 10796 2768 Not Found: /1/
WARNING 2024-02-24 22:42:43,079 basehttp 10796 2768 "GET /1/ HTTP/1.1" 404 2297
WARNING 2024-02-24 22:42:51,698 log 10796 2768 Not Found: /1
WARNING 2024-02-24 22:42:51,699 basehttp 10796 2768 "GET /1 HTTP/1.1" 404 2294
WARNING 2024-02-24 22:43:23,007 log 10796 2768 Not Found: /order_client/1
WARNING 2024-02-24 22:43:23,007 basehttp 10796 2768 "GET /order_client/1 HTTP/1.1" 404 2333
WARNING 2024-02-24 22:43:25,496 log 10796 2768 Not Found: /order_client/1
WARNING 2024-02-24 22:43:25,497 basehttp 10796 2768 "GET /order_client/1 HTTP/1.1" 404 2333
INFO 2024-02-24 22:43:49,288 autoreload 10796 8788 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\shop_project\urls.py changed, reloading.
INFO 2024-02-24 22:43:49,792 autoreload 5908 5604 Watching for file changes with StatReloader
WARNING 2024-02-24 22:43:52,760 log 5908 10548 Not Found: /order_client/1
WARNING 2024-02-24 22:43:52,761 basehttp 5908 10548 "GET /order_client/1 HTTP/1.1" 404 2346
WARNING 2024-02-24 22:43:55,419 log 5908 10548 Not Found: /order_client/1
WARNING 2024-02-24 22:43:55,420 basehttp 5908 10548 "GET /order_client/1 HTTP/1.1" 404 2346
INFO 2024-02-24 22:43:58,334 basehttp 5908 10548 "GET /order_client/ HTTP/1.1" 200 2076
WARNING 2024-02-24 22:43:58,396 basehttp 5908 10548 "GET /static/main/img/Logo_2.png HTTP/1.1" 404 1816
WARNING 2024-02-24 22:44:02,321 log 5908 21048 Not Found: /order_client/2
WARNING 2024-02-24 22:44:02,322 basehttp 5908 21048 "GET /order_client/2 HTTP/1.1" 404 2346
WARNING 2024-02-24 22:44:07,194 log 5908 21048 Not Found: /order_client1
WARNING 2024-02-24 22:44:07,195 basehttp 5908 21048 "GET /order_client1 HTTP/1.1" 404 2343
WARNING 2024-02-24 22:44:12,633 log 5908 21048 Not Found: /order_client-1
WARNING 2024-02-24 22:44:12,634 basehttp 5908 21048 "GET /order_client-1 HTTP/1.1" 404 2346
WARNING 2024-02-24 22:44:18,778 log 5908 21048 Not Found: /order_client/1
WARNING 2024-02-24 22:44:18,779 basehttp 5908 21048 "GET /order_client/1 HTTP/1.1" 404 2346
WARNING 2024-02-24 23:10:27,459 log 5908 21068 Not Found: /order_client/1
WARNING 2024-02-24 23:10:27,460 basehttp 5908 21068 "GET /order_client/1 HTTP/1.1" 404 2346
WARNING 2024-02-24 23:10:44,283 log 5908 21068 Not Found: /order_client/'order_client/<int:client_id>/1
WARNING 2024-02-24 23:10:44,284 basehttp 5908 21068 "GET /order_client/'order_client/%3Cint:client_id%3E/1 HTTP/1.1" 404 2467
WARNING 2024-02-24 23:10:49,735 log 5908 21068 Not Found: /order_client/'order_client/<int:client_id>/
WARNING 2024-02-24 23:10:49,735 basehttp 5908 21068 "GET /order_client/'order_client/%3Cint:client_id%3E/ HTTP/1.1" 404 2464
WARNING 2024-02-24 23:11:14,969 log 5908 21068 Not Found: /order_client/order_client/1/
WARNING 2024-02-24 23:11:14,970 basehttp 5908 21068 "GET /order_client/order_client/1/ HTTP/1.1" 404 2388
WARNING 2024-02-24 23:11:18,161 log 5908 21068 Not Found: /order_client/order_client/1/
WARNING 2024-02-24 23:11:18,162 basehttp 5908 21068 "GET /order_client/order_client/1/ HTTP/1.1" 404 2388
INFO 2024-02-24 23:11:23,855 basehttp 5908 21068 "GET /order_client/ HTTP/1.1" 200 2076
INFO 2024-02-24 23:11:23,873 basehttp 5908 1716 "GET /static/img/logo.png HTTP/1.1" 304 0
INFO 2024-02-24 23:11:23,874 basehttp 5908 21068 "GET /static/css/base.css HTTP/1.1" 304 0
WARNING 2024-02-24 23:11:23,922 basehttp 5908 21068 "GET /static/main/img/Logo_2.png HTTP/1.1" 404 1816
WARNING 2024-02-24 23:15:36,884 log 5908 1716 Not Found: /order_client/client/1
WARNING 2024-02-24 23:15:36,884 basehttp 5908 1716 "GET /order_client/client/1 HTTP/1.1" 404 2367
WARNING 2024-02-24 23:15:41,306 log 5908 1716 Not Found: /order_client/client/1/
WARNING 2024-02-24 23:15:41,307 basehttp 5908 1716 "GET /order_client/client/1/ HTTP/1.1" 404 2370
INFO 2024-02-24 23:16:37,380 autoreload 5908 5604 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py changed, reloading.
INFO 2024-02-24 23:16:38,035 autoreload 12664 1416 Watching for file changes with StatReloader
INFO 2024-02-24 23:20:56,370 autoreload 12664 1416 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\shop_project\urls.py changed, reloading.
INFO 2024-02-24 23:20:57,230 autoreload 19752 20712 Watching for file changes with StatReloader
INFO 2024-02-24 23:21:27,760 autoreload 19752 20712 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\shop_project\urls.py changed, reloading.
INFO 2024-02-24 23:21:28,162 autoreload 2368 11332 Watching for file changes with StatReloader
INFO 2024-02-24 23:21:56,706 basehttp 2368 20932 "GET /order_client/client/1/ HTTP/1.1" 200 2076
WARNING 2024-02-24 23:21:56,815 basehttp 2368 20932 "GET /static/main/img/Logo_2.png HTTP/1.1" 404 1816
WARNING 2024-02-24 23:22:02,914 log 2368 4660 Not Found: /order_client/client/0/
WARNING 2024-02-24 23:22:02,915 basehttp 2368 4660 "GET /order_client/client/0/ HTTP/1.1" 404 2635
INFO 2024-02-24 23:22:10,426 basehttp 2368 4660 "GET /order_client/client/2/ HTTP/1.1" 200 1918
WARNING 2024-02-24 23:22:10,493 basehttp 2368 4660 "GET /static/main/img/Logo_2.png HTTP/1.1" 404 1816
INFO 2024-02-24 23:22:16,151 basehttp 2368 4512 "GET /order_client/client/3/ HTTP/1.1" 200 1918
WARNING 2024-02-24 23:22:16,211 basehttp 2368 4512 "GET /static/main/img/Logo_2.png HTTP/1.1" 404 1816
INFO 2024-02-24 23:22:21,487 basehttp 2368 9052 "GET /order_client/client/1/ HTTP/1.1" 200 2076
WARNING 2024-02-24 23:22:21,530 basehttp 2368 9052 "GET /static/main/img/Logo_2.png HTTP/1.1" 404 1816
INFO 2024-02-24 23:22:46,643 autoreload 2368 11332 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\shop_project\urls.py changed, reloading.
INFO 2024-02-24 23:22:47,036 autoreload 3632 20072 Watching for file changes with StatReloader
INFO 2024-02-26 00:55:59,159 autoreload 24124 10340 Watching for file changes with StatReloader
INFO 2024-02-26 00:57:06,890 autoreload 20592 23616 Watching for file changes with StatReloader
INFO 2024-02-26 00:58:18,846 autoreload 3504 23836 Watching for file changes with StatReloader
INFO 2024-02-26 00:58:50,503 autoreload 3504 23836 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\shop_project\settings.py changed, reloading.
INFO 2024-02-26 00:58:50,911 autoreload 23892 14548 Watching for file changes with StatReloader
INFO 2024-02-26 00:59:36,705 autoreload 23892 14548 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\shop_project\settings.py changed, reloading.
INFO 2024-02-26 00:59:37,108 autoreload 21704 9608 Watching for file changes with StatReloader
INFO 2024-02-26 00:59:57,117 autoreload 3236 15800 Watching for file changes with StatReloader
INFO 2024-02-26 01:00:30,493 autoreload 22152 8676 Watching for file changes with StatReloader
WARNING 2024-02-26 01:00:33,469 log 22152 23716 Not Found: /
WARNING 2024-02-26 01:00:33,469 basehttp 22152 23716 "GET / HTTP/1.1" 404 2399
ERROR 2024-02-26 01:01:31,546 log 22152 23716 Internal Server Error: /edit_product/product/1/
Traceback (most recent call last):
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\backends\utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\backends\sqlite3\base.py", line 329, in execute
    return super().execute(query, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
sqlite3.OperationalError: no such column: myapp_2_product.photo

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py", line 39, in add_or_edit_product
    product = Product.objects.get(pk=product_id)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\models\manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\models\query.py", line 645, in get
    num = len(clone)
          ^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\models\query.py", line 382, in __len__
    self._fetch_all()
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\models\query.py", line 1928, in _fetch_all
    self._result_cache = list(self._iterable_class(self))
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\models\query.py", line 91, in __iter__
    results = compiler.execute_sql(
              ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\models\sql\compiler.py", line 1562, in execute_sql
    cursor.execute(sql, params)
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\backends\utils.py", line 122, in execute
    return super().execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\backends\utils.py", line 79, in execute
    return self._execute_with_wrappers(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\backends\utils.py", line 92, in _execute_with_wrappers
    return executor(sql, params, many, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\backends\utils.py", line 100, in _execute
    with self.db.wrap_database_errors:
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\backends\utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\backends\sqlite3\base.py", line 329, in execute
    return super().execute(query, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django.db.utils.OperationalError: no such column: myapp_2_product.photo
ERROR 2024-02-26 01:01:31,556 basehttp 22152 23716 "GET /edit_product/product/1/ HTTP/1.1" 500 138400
ERROR 2024-02-26 01:01:59,757 log 22152 23716 Internal Server Error: /edit_product/product/1/
Traceback (most recent call last):
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\backends\utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\backends\sqlite3\base.py", line 329, in execute
    return super().execute(query, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
sqlite3.OperationalError: no such column: myapp_2_product.photo

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py", line 39, in add_or_edit_product
    product = Product.objects.get(pk=product_id)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\models\manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\models\query.py", line 645, in get
    num = len(clone)
          ^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\models\query.py", line 382, in __len__
    self._fetch_all()
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\models\query.py", line 1928, in _fetch_all
    self._result_cache = list(self._iterable_class(self))
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\models\query.py", line 91, in __iter__
    results = compiler.execute_sql(
              ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\models\sql\compiler.py", line 1562, in execute_sql
    cursor.execute(sql, params)
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\backends\utils.py", line 122, in execute
    return super().execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\backends\utils.py", line 79, in execute
    return self._execute_with_wrappers(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\backends\utils.py", line 92, in _execute_with_wrappers
    return executor(sql, params, many, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\backends\utils.py", line 100, in _execute
    with self.db.wrap_database_errors:
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\backends\utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\backends\sqlite3\base.py", line 329, in execute
    return super().execute(query, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django.db.utils.OperationalError: no such column: myapp_2_product.photo
ERROR 2024-02-26 01:01:59,762 basehttp 22152 23716 "GET /edit_product/product/1/ HTTP/1.1" 500 138537
INFO 2024-02-26 01:07:46,994 autoreload 22152 8676 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\models.py changed, reloading.
INFO 2024-02-26 01:07:47,637 autoreload 21908 23644 Watching for file changes with StatReloader
ERROR 2024-02-26 01:08:28,046 log 21908 23576 Internal Server Error: /edit_product/product/1/
Traceback (most recent call last):
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\backends\utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\backends\sqlite3\base.py", line 329, in execute
    return super().execute(query, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
sqlite3.OperationalError: no such column: myapp_2_product.photo

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py", line 39, in add_or_edit_product
    product = Product.objects.get(pk=product_id)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\models\manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\models\query.py", line 645, in get
    num = len(clone)
          ^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\models\query.py", line 382, in __len__
    self._fetch_all()
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\models\query.py", line 1928, in _fetch_all
    self._result_cache = list(self._iterable_class(self))
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\models\query.py", line 91, in __iter__
    results = compiler.execute_sql(
              ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\models\sql\compiler.py", line 1562, in execute_sql
    cursor.execute(sql, params)
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\backends\utils.py", line 122, in execute
    return super().execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\backends\utils.py", line 79, in execute
    return self._execute_with_wrappers(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\backends\utils.py", line 92, in _execute_with_wrappers
    return executor(sql, params, many, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\backends\utils.py", line 100, in _execute
    with self.db.wrap_database_errors:
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\backends\utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\backends\sqlite3\base.py", line 329, in execute
    return super().execute(query, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django.db.utils.OperationalError: no such column: myapp_2_product.photo
ERROR 2024-02-26 01:08:28,050 basehttp 21908 23576 "GET /edit_product/product/1/ HTTP/1.1" 500 138537
INFO 2024-02-26 01:09:10,007 autoreload 21908 23644 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py changed, reloading.
INFO 2024-02-26 01:09:10,456 autoreload 17008 20124 Watching for file changes with StatReloader
INFO 2024-02-26 01:09:19,696 autoreload 17008 20124 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py changed, reloading.
INFO 2024-02-26 01:09:20,079 autoreload 19304 23964 Watching for file changes with StatReloader
INFO 2024-02-26 01:09:23,335 autoreload 19304 23964 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py changed, reloading.
INFO 2024-02-26 01:09:23,909 autoreload 23532 23488 Watching for file changes with StatReloader
INFO 2024-02-26 01:09:30,108 autoreload 23532 23488 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py changed, reloading.
INFO 2024-02-26 01:09:30,530 autoreload 13432 23352 Watching for file changes with StatReloader
INFO 2024-02-26 01:09:38,855 autoreload 13432 23352 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py changed, reloading.
INFO 2024-02-26 01:09:39,315 autoreload 21872 19188 Watching for file changes with StatReloader
INFO 2024-02-26 01:09:53,593 autoreload 21872 19188 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py changed, reloading.
INFO 2024-02-26 01:09:54,028 autoreload 15200 17700 Watching for file changes with StatReloader
INFO 2024-02-26 01:10:03,367 autoreload 15200 17700 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py changed, reloading.
INFO 2024-02-26 01:10:03,867 autoreload 2928 23872 Watching for file changes with StatReloader
INFO 2024-02-26 01:10:08,128 autoreload 2928 23872 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py changed, reloading.
INFO 2024-02-26 01:10:08,742 autoreload 22612 9448 Watching for file changes with StatReloader
INFO 2024-02-26 01:10:26,239 autoreload 22612 9448 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py changed, reloading.
INFO 2024-02-26 01:10:26,740 autoreload 23576 22312 Watching for file changes with StatReloader
INFO 2024-02-26 01:10:36,079 autoreload 23576 22312 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py changed, reloading.
INFO 2024-02-26 01:10:36,599 autoreload 18952 3168 Watching for file changes with StatReloader
INFO 2024-02-26 01:11:17,448 autoreload 18952 3168 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py changed, reloading.
INFO 2024-02-26 01:11:17,894 autoreload 8804 1644 Watching for file changes with StatReloader
ERROR 2024-02-26 01:11:18,460 log 8804 22056 Internal Server Error: /edit_product/product/1/
Traceback (most recent call last):
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\backends\utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\backends\sqlite3\base.py", line 329, in execute
    return super().execute(query, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
sqlite3.OperationalError: no such column: myapp_2_product.photo

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py", line 39, in add_or_edit_product
    product = Product.objects.get(pk=product_id)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\models\manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\models\query.py", line 645, in get
    num = len(clone)
          ^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\models\query.py", line 382, in __len__
    self._fetch_all()
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\models\query.py", line 1928, in _fetch_all
    self._result_cache = list(self._iterable_class(self))
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\models\query.py", line 91, in __iter__
    results = compiler.execute_sql(
              ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\models\sql\compiler.py", line 1562, in execute_sql
    cursor.execute(sql, params)
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\backends\utils.py", line 122, in execute
    return super().execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\backends\utils.py", line 79, in execute
    return self._execute_with_wrappers(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\backends\utils.py", line 92, in _execute_with_wrappers
    return executor(sql, params, many, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\backends\utils.py", line 100, in _execute
    with self.db.wrap_database_errors:
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\backends\utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\backends\sqlite3\base.py", line 329, in execute
    return super().execute(query, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django.db.utils.OperationalError: no such column: myapp_2_product.photo
ERROR 2024-02-26 01:11:18,465 basehttp 8804 22056 "GET /edit_product/product/1/ HTTP/1.1" 500 138537
INFO 2024-02-26 01:11:59,737 autoreload 8804 1644 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\urls.py changed, reloading.
INFO 2024-02-26 01:12:00,230 autoreload 19544 24272 Watching for file changes with StatReloader
ERROR 2024-02-26 01:12:01,285 log 19544 16216 Internal Server Error: /edit_product/product/1/
Traceback (most recent call last):
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\backends\utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\backends\sqlite3\base.py", line 329, in execute
    return super().execute(query, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
sqlite3.OperationalError: no such column: myapp_2_product.photo

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py", line 39, in add_or_edit_product
    product = Product.objects.get(pk=product_id)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\models\manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\models\query.py", line 645, in get
    num = len(clone)
          ^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\models\query.py", line 382, in __len__
    self._fetch_all()
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\models\query.py", line 1928, in _fetch_all
    self._result_cache = list(self._iterable_class(self))
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\models\query.py", line 91, in __iter__
    results = compiler.execute_sql(
              ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\models\sql\compiler.py", line 1562, in execute_sql
    cursor.execute(sql, params)
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\backends\utils.py", line 122, in execute
    return super().execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\backends\utils.py", line 79, in execute
    return self._execute_with_wrappers(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\backends\utils.py", line 92, in _execute_with_wrappers
    return executor(sql, params, many, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\backends\utils.py", line 100, in _execute
    with self.db.wrap_database_errors:
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\backends\utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\backends\sqlite3\base.py", line 329, in execute
    return super().execute(query, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django.db.utils.OperationalError: no such column: myapp_2_product.photo
ERROR 2024-02-26 01:12:01,290 basehttp 19544 16216 "GET /edit_product/product/1/ HTTP/1.1" 500 138537
ERROR 2024-02-26 01:12:02,103 log 19544 16216 Internal Server Error: /edit_product/product/1/
Traceback (most recent call last):
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\backends\utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\backends\sqlite3\base.py", line 329, in execute
    return super().execute(query, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
sqlite3.OperationalError: no such column: myapp_2_product.photo

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py", line 39, in add_or_edit_product
    product = Product.objects.get(pk=product_id)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\models\manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\models\query.py", line 645, in get
    num = len(clone)
          ^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\models\query.py", line 382, in __len__
    self._fetch_all()
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\models\query.py", line 1928, in _fetch_all
    self._result_cache = list(self._iterable_class(self))
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\models\query.py", line 91, in __iter__
    results = compiler.execute_sql(
              ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\models\sql\compiler.py", line 1562, in execute_sql
    cursor.execute(sql, params)
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\backends\utils.py", line 122, in execute
    return super().execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\backends\utils.py", line 79, in execute
    return self._execute_with_wrappers(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\backends\utils.py", line 92, in _execute_with_wrappers
    return executor(sql, params, many, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\backends\utils.py", line 100, in _execute
    with self.db.wrap_database_errors:
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\backends\utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\backends\sqlite3\base.py", line 329, in execute
    return super().execute(query, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django.db.utils.OperationalError: no such column: myapp_2_product.photo
ERROR 2024-02-26 01:12:02,108 basehttp 19544 16216 "GET /edit_product/product/1/ HTTP/1.1" 500 138537
INFO 2024-02-26 01:14:02,991 autoreload 19544 24272 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py changed, reloading.
INFO 2024-02-26 01:14:03,442 autoreload 18588 24468 Watching for file changes with StatReloader
INFO 2024-02-26 01:14:09,677 autoreload 18588 24468 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py changed, reloading.
INFO 2024-02-26 01:14:10,138 autoreload 24096 23676 Watching for file changes with StatReloader
INFO 2024-02-26 01:14:26,467 autoreload 24096 23676 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py changed, reloading.
INFO 2024-02-26 01:14:26,987 autoreload 20776 23460 Watching for file changes with StatReloader
ERROR 2024-02-26 01:14:27,536 log 20776 23420 Internal Server Error: /edit_product/product/1/
Traceback (most recent call last):
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\backends\utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\backends\sqlite3\base.py", line 329, in execute
    return super().execute(query, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
sqlite3.OperationalError: no such column: myapp_2_product.photo

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py", line 40, in add_or_edit_product
    product = Product.objects.get(pk=product_id)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\models\manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\models\query.py", line 645, in get
    num = len(clone)
          ^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\models\query.py", line 382, in __len__
    self._fetch_all()
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\models\query.py", line 1928, in _fetch_all
    self._result_cache = list(self._iterable_class(self))
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\models\query.py", line 91, in __iter__
    results = compiler.execute_sql(
              ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\models\sql\compiler.py", line 1562, in execute_sql
    cursor.execute(sql, params)
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\backends\utils.py", line 122, in execute
    return super().execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\backends\utils.py", line 79, in execute
    return self._execute_with_wrappers(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\backends\utils.py", line 92, in _execute_with_wrappers
    return executor(sql, params, many, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\backends\utils.py", line 100, in _execute
    with self.db.wrap_database_errors:
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\backends\utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\backends\sqlite3\base.py", line 329, in execute
    return super().execute(query, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django.db.utils.OperationalError: no such column: myapp_2_product.photo
ERROR 2024-02-26 01:14:27,541 basehttp 20776 23420 "GET /edit_product/product/1/ HTTP/1.1" 500 138537
INFO 2024-02-26 01:17:06,325 autoreload 19664 15004 Watching for file changes with StatReloader
INFO 2024-02-26 01:17:06,500 basehttp 19664 19212 "GET /edit_product/product/1/ HTTP/1.1" 200 3213
INFO 2024-02-26 01:17:06,548 basehttp 19664 19212 "GET /static/img/logo.png HTTP/1.1" 200 7903
INFO 2024-02-26 01:17:06,550 basehttp 19664 19212 "GET /static/css/base.css HTTP/1.1" 304 0
WARNING 2024-02-26 01:17:06,675 basehttp 19664 19212 "GET /static/main/img/Logo_2.png HTTP/1.1" 404 1816
WARNING 2024-02-26 01:17:08,595 log 19664 14768 Not Found: /
WARNING 2024-02-26 01:17:08,596 basehttp 19664 14768 "GET / HTTP/1.1" 404 2399
ERROR 2024-02-26 01:19:24,138 log 19664 14768 Internal Server Error: /edit_product/product/1/
Traceback (most recent call last):
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py", line 47, in add_or_edit_product
    image = form.cleaned_data['image']
            ~~~~~~~~~~~~~~~~~^^^^^^^^^
KeyError: 'image'
ERROR 2024-02-26 01:19:24,143 basehttp 19664 14768 "POST /edit_product/product/1/ HTTP/1.1" 500 69135
INFO 2024-02-26 01:20:44,859 autoreload 19664 15004 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py changed, reloading.
INFO 2024-02-26 01:20:45,432 autoreload 3580 16512 Watching for file changes with StatReloader
ERROR 2024-02-26 01:20:47,556 log 3580 19072 Internal Server Error: /edit_product/product/1/
Traceback (most recent call last):
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py", line 47, in add_or_edit_product
    image = form.cleaned_data['image']
            ~~~~~~~~~~~~~~~~~^^^^^^^^^
KeyError: 'image'
ERROR 2024-02-26 01:20:47,557 basehttp 3580 19072 "POST /edit_product/product/1/ HTTP/1.1" 500 69128
INFO 2024-02-26 01:20:53,984 autoreload 3580 16512 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py changed, reloading.
INFO 2024-02-26 01:20:54,531 autoreload 18296 22216 Watching for file changes with StatReloader
INFO 2024-02-26 01:23:43,979 autoreload 18296 22216 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py changed, reloading.
INFO 2024-02-26 01:23:44,693 autoreload 18520 13556 Watching for file changes with StatReloader
ERROR 2024-02-26 01:23:45,772 log 18520 20124 Internal Server Error: /edit_product/product/1/
Traceback (most recent call last):
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py", line 47, in add_or_edit_product
    photo = form.cleaned_data['image']
            ~~~~~~~~~~~~~~~~~^^^^^^^^^
KeyError: 'image'
ERROR 2024-02-26 01:23:45,774 basehttp 18520 20124 "POST /edit_product/product/1/ HTTP/1.1" 500 69135
ERROR 2024-02-26 01:23:50,940 log 18520 20124 Internal Server Error: /edit_product/product/1/
Traceback (most recent call last):
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py", line 47, in add_or_edit_product
    photo = form.cleaned_data['image']
            ~~~~~~~~~~~~~~~~~^^^^^^^^^
KeyError: 'image'
ERROR 2024-02-26 01:23:50,941 basehttp 18520 20124 "POST /edit_product/product/1/ HTTP/1.1" 500 69135
INFO 2024-02-26 01:24:11,469 autoreload 18520 13556 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py changed, reloading.
INFO 2024-02-26 01:24:11,944 autoreload 18192 17140 Watching for file changes with StatReloader
ERROR 2024-02-26 01:24:13,098 log 18192 1644 Internal Server Error: /edit_product/product/1/
Traceback (most recent call last):
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py", line 50, in add_or_edit_product
    return redirect('product_list')
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\shortcuts.py", line 49, in redirect
    return redirect_class(resolve_url(to, *args, **kwargs))
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\shortcuts.py", line 180, in resolve_url
    return reverse(to, args=args, kwargs=kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\urls\base.py", line 88, in reverse
    return resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\urls\resolvers.py", line 851, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'product_list' not found. 'product_list' is not a valid view function or pattern name.
ERROR 2024-02-26 01:24:13,102 basehttp 18192 1644 "POST /edit_product/product/1/ HTTP/1.1" 500 88393
INFO 2024-02-26 01:27:14,117 autoreload 18192 17140 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py changed, reloading.
INFO 2024-02-26 01:27:14,929 autoreload 18296 20564 Watching for file changes with StatReloader
ERROR 2024-02-26 01:27:35,484 log 18296 18588 Internal Server Error: /edit_product/product/1/
Traceback (most recent call last):
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py", line 51, in add_or_edit_product
    return redirect('product_list')
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\shortcuts.py", line 49, in redirect
    return redirect_class(resolve_url(to, *args, **kwargs))
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\shortcuts.py", line 180, in resolve_url
    return reverse(to, args=args, kwargs=kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\urls\base.py", line 88, in reverse
    return resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\urls\resolvers.py", line 851, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'product_list' not found. 'product_list' is not a valid view function or pattern name.
ERROR 2024-02-26 01:27:35,487 basehttp 18296 18588 "POST /edit_product/product/1/ HTTP/1.1" 500 87938
INFO 2024-02-26 01:29:40,310 autoreload 18296 20564 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py changed, reloading.
INFO 2024-02-26 01:29:40,808 autoreload 23116 18896 Watching for file changes with StatReloader
INFO 2024-02-26 01:29:42,044 basehttp 23116 13336 "POST /edit_product/product/1/ HTTP/1.1" 200 3310
WARNING 2024-02-26 01:29:42,097 basehttp 23116 13336 "GET /static/main/img/Logo_2.png HTTP/1.1" 404 1816
INFO 2024-02-26 01:29:48,715 basehttp 23116 18808 "POST /edit_product/product/1/ HTTP/1.1" 200 3310
WARNING 2024-02-26 01:29:48,755 basehttp 23116 18808 "GET /static/main/img/Logo_2.png HTTP/1.1" 404 1816
INFO 2024-02-26 01:29:49,891 basehttp 23116 22220 "POST /edit_product/product/1/ HTTP/1.1" 200 3310
WARNING 2024-02-26 01:29:49,931 basehttp 23116 22220 "GET /static/main/img/Logo_2.png HTTP/1.1" 404 1816
WARNING 2024-02-26 01:29:52,332 log 23116 23296 Not Found: /media/mango-shake-fresh-tropical-fruit-smoothies_7sjnVvj.jpg
WARNING 2024-02-26 01:29:52,332 basehttp 23116 23296 "GET /media/mango-shake-fresh-tropical-fruit-smoothies_7sjnVvj.jpg HTTP/1.1" 404 2597
INFO 2024-02-26 01:30:52,919 autoreload 23116 18896 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py changed, reloading.
INFO 2024-02-26 01:30:53,566 autoreload 19348 6656 Watching for file changes with StatReloader
INFO 2024-02-26 01:32:17,022 autoreload 19348 6656 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py changed, reloading.
INFO 2024-02-26 01:32:17,413 autoreload 15348 20056 Watching for file changes with StatReloader
INFO 2024-02-26 01:32:35,770 autoreload 15348 20056 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py changed, reloading.
INFO 2024-02-26 01:32:36,147 autoreload 20564 18296 Watching for file changes with StatReloader
WARNING 2024-02-26 01:32:38,097 log 20564 2928 Not Found: /
WARNING 2024-02-26 01:32:38,099 basehttp 20564 2928 "GET / HTTP/1.1" 404 2399
INFO 2024-02-26 01:32:45,288 basehttp 20564 2928 "POST /edit_product/product/1/ HTTP/1.1" 200 3310
WARNING 2024-02-26 01:32:45,334 basehttp 20564 2928 "GET /static/main/img/Logo_2.png HTTP/1.1" 404 1816
INFO 2024-02-26 01:32:48,850 basehttp 20564 10872 "POST /edit_product/product/1/ HTTP/1.1" 200 3310
WARNING 2024-02-26 01:32:48,891 basehttp 20564 10872 "GET /static/main/img/Logo_2.png HTTP/1.1" 404 1816
INFO 2024-02-26 01:32:51,256 basehttp 20564 21968 "POST /edit_product/product/1/ HTTP/1.1" 200 3310
WARNING 2024-02-26 01:32:51,292 basehttp 20564 21968 "GET /static/main/img/Logo_2.png HTTP/1.1" 404 1816
INFO 2024-02-26 01:33:05,088 basehttp 20564 23240 "GET /edit_product/product/2/ HTTP/1.1" 200 3149
WARNING 2024-02-26 01:33:05,154 basehttp 20564 23240 "GET /static/main/img/Logo_2.png HTTP/1.1" 404 1816
ERROR 2024-02-26 01:33:08,362 log 20564 5928 Internal Server Error: /edit_product/product/2/
Traceback (most recent call last):
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py", line 49, in add_or_edit_product
    fs.save(photo.name, photo)
            ^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'name'
ERROR 2024-02-26 01:33:08,363 basehttp 20564 5928 "POST /edit_product/product/2/ HTTP/1.1" 500 69430
ERROR 2024-02-26 01:33:21,215 log 20564 5928 Internal Server Error: /edit_product/product/2/
Traceback (most recent call last):
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py", line 49, in add_or_edit_product
    fs.save(photo.name, photo)
            ^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'name'
ERROR 2024-02-26 01:33:21,218 basehttp 20564 5928 "POST /edit_product/product/2/ HTTP/1.1" 500 69430
INFO 2024-02-26 01:34:48,502 autoreload 20564 18296 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py changed, reloading.
INFO 2024-02-26 01:34:48,969 autoreload 17148 13832 Watching for file changes with StatReloader
INFO 2024-02-26 01:35:03,411 autoreload 17148 13832 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py changed, reloading.
INFO 2024-02-26 01:35:03,923 autoreload 18636 20916 Watching for file changes with StatReloader
INFO 2024-02-26 01:35:10,094 autoreload 18636 20916 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py changed, reloading.
INFO 2024-02-26 01:35:10,532 autoreload 13356 2868 Watching for file changes with StatReloader
INFO 2024-02-26 01:35:16,825 autoreload 13356 2868 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py changed, reloading.
INFO 2024-02-26 01:35:17,259 autoreload 15348 22224 Watching for file changes with StatReloader
INFO 2024-02-26 01:35:42,920 autoreload 15348 22224 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py changed, reloading.
INFO 2024-02-26 01:35:43,393 autoreload 7176 15772 Watching for file changes with StatReloader
INFO 2024-02-26 01:35:50,707 autoreload 7176 15772 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py changed, reloading.
INFO 2024-02-26 01:35:51,240 autoreload 13024 4480 Watching for file changes with StatReloader
INFO 2024-02-26 01:36:42,316 autoreload 13024 4480 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py changed, reloading.
INFO 2024-02-26 01:36:42,780 autoreload 19440 6756 Watching for file changes with StatReloader
INFO 2024-02-26 01:37:02,140 basehttp 19440 3576 "POST /edit_product/product/2/ HTTP/1.1" 200 3153
WARNING 2024-02-26 01:37:02,187 basehttp 19440 3576 "GET /static/main/img/Logo_2.png HTTP/1.1" 404 1816
ERROR 2024-02-26 01:37:21,669 log 19440 15120 Internal Server Error: /edit_product/product/2/
Traceback (most recent call last):
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py", line 53, in add_or_edit_product
    fs.save(photo.name, photo)
            ^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'name'
ERROR 2024-02-26 01:37:21,672 basehttp 19440 15120 "POST /edit_product/product/2/ HTTP/1.1" 500 69620
INFO 2024-02-26 01:38:05,163 autoreload 19440 6756 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py changed, reloading.
INFO 2024-02-26 01:38:05,737 autoreload 19832 14712 Watching for file changes with StatReloader
ERROR 2024-02-26 01:38:06,621 log 19832 22816 Internal Server Error: /edit_product/product/2/
Traceback (most recent call last):
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py", line 53, in add_or_edit_product
    fs.save(photo)
TypeError: Storage.save() missing 1 required positional argument: 'content'
ERROR 2024-02-26 01:38:06,623 basehttp 19832 22816 "POST /edit_product/product/2/ HTTP/1.1" 500 69605
ERROR 2024-02-26 01:38:18,170 log 19832 22816 Internal Server Error: /edit_product/product/2/
Traceback (most recent call last):
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py", line 53, in add_or_edit_product
    fs.save(photo)
TypeError: Storage.save() missing 1 required positional argument: 'content'
ERROR 2024-02-26 01:38:18,172 basehttp 19832 22816 "POST /edit_product/product/2/ HTTP/1.1" 500 69605
INFO 2024-02-26 01:39:11,408 autoreload 19832 14712 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py changed, reloading.
INFO 2024-02-26 01:39:12,127 autoreload 8892 23928 Watching for file changes with StatReloader
INFO 2024-02-26 01:40:07,336 autoreload 8892 23928 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py changed, reloading.
INFO 2024-02-26 01:40:08,214 autoreload 18592 7704 Watching for file changes with StatReloader
INFO 2024-02-26 01:40:20,490 autoreload 18592 7704 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py changed, reloading.
INFO 2024-02-26 01:40:20,967 autoreload 22544 3292 Watching for file changes with StatReloader
INFO 2024-02-26 01:40:28,275 autoreload 22544 3292 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py changed, reloading.
INFO 2024-02-26 01:40:29,024 autoreload 1840 24500 Watching for file changes with StatReloader
INFO 2024-02-26 01:40:40,455 autoreload 1840 24500 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\views.py changed, reloading.
INFO 2024-02-26 01:40:40,955 autoreload 4420 24168 Watching for file changes with StatReloader
INFO 2024-02-26 01:40:45,336 basehttp 4420 20700 "POST /edit_product/product/2/ HTTP/1.1" 200 3153
WARNING 2024-02-26 01:40:45,393 basehttp 4420 20700 "GET /static/main/img/Logo_2.png HTTP/1.1" 404 1816
INFO 2024-02-26 01:40:46,640 basehttp 4420 22312 "POST /edit_product/product/2/ HTTP/1.1" 200 3153
WARNING 2024-02-26 01:40:46,687 basehttp 4420 22312 "GET /static/main/img/Logo_2.png HTTP/1.1" 404 1816
INFO 2024-02-26 01:40:49,195 basehttp 4420 22480 "POST /edit_product/product/2/ HTTP/1.1" 200 3153
WARNING 2024-02-26 01:40:49,238 basehttp 4420 22480 "GET /static/main/img/Logo_2.png HTTP/1.1" 404 1816
INFO 2024-02-26 01:40:50,073 basehttp 4420 19780 "POST /edit_product/product/2/ HTTP/1.1" 200 3153
WARNING 2024-02-26 01:40:50,114 basehttp 4420 19780 "GET /static/main/img/Logo_2.png HTTP/1.1" 404 1816
INFO 2024-02-26 01:41:49,718 basehttp 4420 17428 "POST /edit_product/product/2/ HTTP/1.1" 200 3153
WARNING 2024-02-26 01:41:49,766 basehttp 4420 17428 "GET /static/main/img/Logo_2.png HTTP/1.1" 404 1816
INFO 2024-02-29 21:17:33,219 autoreload 15920 20240 Watching for file changes with StatReloader
WARNING 2024-02-29 21:17:35,278 log 15920 6596 Not Found: /
WARNING 2024-02-29 21:17:35,318 basehttp 15920 6596 "GET / HTTP/1.1" 404 2399
INFO 2024-02-29 21:17:53,242 basehttp 15920 6596 "GET /admin/ HTTP/1.1" 302 0
INFO 2024-02-29 21:17:53,565 basehttp 15920 6596 "GET /admin/login/?next=/admin/ HTTP/1.1" 200 4359
INFO 2024-02-29 21:17:54,012 basehttp 15920 6752 "GET /static/admin/css/nav_sidebar.css HTTP/1.1" 200 2810
INFO 2024-02-29 21:17:54,039 basehttp 15920 17848 "GET /static/admin/css/login.css HTTP/1.1" 200 958
INFO 2024-02-29 21:17:54,042 basehttp 15920 6596 "GET /static/admin/css/base.css HTTP/1.1" 200 21544
INFO 2024-02-29 21:17:54,050 basehttp 15920 21236 "GET /static/admin/css/dark_mode.css HTTP/1.1" 200 2682
INFO 2024-02-29 21:17:54,053 basehttp 15920 6752 "GET /static/admin/js/nav_sidebar.js HTTP/1.1" 200 3063
INFO 2024-02-29 21:17:54,067 basehttp 15920 24636 "GET /static/admin/css/responsive.css HTTP/1.1" 200 17905
INFO 2024-02-29 21:17:54,076 basehttp 15920 14512 "GET /static/admin/js/theme.js HTTP/1.1" 200 1943
INFO 2024-02-29 21:18:04,691 basehttp 15920 14512 "POST /admin/login/?next=/admin/ HTTP/1.1" 302 0
INFO 2024-02-29 21:18:04,770 basehttp 15920 14512 "GET /admin/ HTTP/1.1" 200 5944
INFO 2024-02-29 21:18:04,806 basehttp 15920 14512 "GET /static/admin/css/dashboard.css HTTP/1.1" 200 441
INFO 2024-02-29 21:18:04,875 basehttp 15920 24636 "GET /static/admin/img/icon-changelink.svg HTTP/1.1" 200 380
INFO 2024-02-29 21:18:04,875 basehttp 15920 14512 "GET /static/admin/img/icon-addlink.svg HTTP/1.1" 200 331
INFO 2024-02-29 21:19:51,399 autoreload 15920 20240 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\models.py changed, reloading.
INFO 2024-02-29 21:19:52,602 autoreload 16464 21500 Watching for file changes with StatReloader
INFO 2024-02-29 21:21:23,543 autoreload 16464 21500 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\admin.py changed, reloading.
INFO 2024-02-29 21:21:24,621 autoreload 9564 23188 Watching for file changes with StatReloader
INFO 2024-02-29 21:21:41,605 autoreload 9564 23188 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\admin.py changed, reloading.
INFO 2024-02-29 21:21:42,775 autoreload 14688 5528 Watching for file changes with StatReloader
INFO 2024-02-29 22:02:26,509 autoreload 25884 18624 Watching for file changes with StatReloader
INFO 2024-02-29 22:10:10,200 autoreload 7220 21044 Watching for file changes with StatReloader
WARNING 2024-02-29 22:10:12,293 log 7220 13856 Not Found: /
WARNING 2024-02-29 22:10:12,293 basehttp 7220 13856 "GET / HTTP/1.1" 404 2399
INFO 2024-02-29 22:10:22,886 basehttp 7220 13856 "GET /admin/ HTTP/1.1" 200 6633
INFO 2024-02-29 22:10:56,694 autoreload 7220 21044 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\admin.py changed, reloading.
INFO 2024-02-29 22:10:57,111 autoreload 18000 22332 Watching for file changes with StatReloader
INFO 2024-02-29 22:11:01,499 basehttp 18000 5344 "GET /admin/ HTTP/1.1" 200 7525
INFO 2024-02-29 22:11:06,552 basehttp 18000 5344 "GET /admin/myapp_2/client/add/ HTTP/1.1" 200 12140
INFO 2024-02-29 22:11:06,595 basehttp 18000 26048 "GET /admin/jsi18n/ HTTP/1.1" 200 17546
INFO 2024-02-29 22:11:06,662 basehttp 18000 5344 "GET /static/admin/css/forms.css HTTP/1.1" 200 9090
INFO 2024-02-29 22:11:06,673 basehttp 18000 14188 "GET /static/admin/js/jquery.init.js HTTP/1.1" 200 347
INFO 2024-02-29 22:11:06,681 basehttp 18000 5344 "GET /static/admin/js/admin/RelatedObjectLookups.js HTTP/1.1" 200 9042
INFO 2024-02-29 22:11:06,683 basehttp 18000 14188 "GET /static/admin/css/widgets.css HTTP/1.1" 200 11800
INFO 2024-02-29 22:11:06,683 basehttp 18000 25764 "GET /static/admin/js/calendar.js HTTP/1.1" 200 9141
INFO 2024-02-29 22:11:06,686 basehttp 18000 12112 "GET /static/admin/js/admin/DateTimeShortcuts.js HTTP/1.1" 200 19319
INFO 2024-02-29 22:11:06,690 basehttp 18000 26048 "GET /static/admin/js/core.js HTTP/1.1" 200 6208
INFO 2024-02-29 22:11:06,698 basehttp 18000 5344 "GET /static/admin/js/actions.js HTTP/1.1" 200 8067
INFO 2024-02-29 22:11:06,698 basehttp 18000 14188 "GET /static/admin/js/urlify.js HTTP/1.1" 200 7887
INFO 2024-02-29 22:11:06,698 basehttp 18000 25764 "GET /static/admin/js/prepopulate.js HTTP/1.1" 200 1531
INFO 2024-02-29 22:11:06,705 basehttp 18000 26048 "GET /static/admin/js/prepopulate_init.js HTTP/1.1" 200 586
INFO 2024-02-29 22:11:06,722 basehttp 18000 17132 "GET /static/admin/js/vendor/jquery/jquery.js HTTP/1.1" 200 285314
INFO 2024-02-29 22:11:06,739 basehttp 18000 17132 "GET /static/admin/js/change_form.js HTTP/1.1" 200 606
INFO 2024-02-29 22:11:06,757 basehttp 18000 12112 "GET /static/admin/js/vendor/xregexp/xregexp.js HTTP/1.1" 200 325171
INFO 2024-02-29 22:11:06,793 basehttp 18000 12112 "GET /static/admin/img/icon-calendar.svg HTTP/1.1" 200 1086
INFO 2024-02-29 22:11:10,018 basehttp 18000 12112 "GET /admin/ HTTP/1.1" 200 7525
INFO 2024-02-29 22:11:12,340 basehttp 18000 12112 "GET /admin/myapp_2/client/add/ HTTP/1.1" 200 12140
INFO 2024-02-29 22:11:12,355 basehttp 18000 12112 "GET /admin/jsi18n/ HTTP/1.1" 200 17546
INFO 2024-02-29 22:11:15,125 basehttp 18000 12112 "GET /admin/ HTTP/1.1" 200 7525
INFO 2024-02-29 22:11:17,222 basehttp 18000 12112 "GET /admin/myapp_2/order/add/ HTTP/1.1" 200 15371
INFO 2024-02-29 22:11:17,236 basehttp 18000 17132 "GET /admin/jsi18n/ HTTP/1.1" 200 17546
INFO 2024-02-29 22:11:17,238 basehttp 18000 12112 "GET /static/admin/img/icon-viewlink.svg HTTP/1.1" 200 581
INFO 2024-02-29 22:11:42,602 basehttp 18000 12112 "GET /admin/ HTTP/1.1" 200 7525
INFO 2024-02-29 22:11:45,714 basehttp 18000 12112 "GET /admin/myapp_2/order/add/ HTTP/1.1" 200 15371
INFO 2024-02-29 22:11:45,728 basehttp 18000 12112 "GET /admin/jsi18n/ HTTP/1.1" 200 17546
INFO 2024-02-29 22:12:01,870 basehttp 18000 12112 "GET /admin/myapp_2/product/add/?_to_field=id&_popup=1 HTTP/1.1" 200 7733
INFO 2024-02-29 22:12:01,882 basehttp 18000 12112 "GET /admin/jsi18n/ HTTP/1.1" 200 17546
INFO 2024-02-29 22:12:18,312 basehttp 18000 12112 "GET /admin/ HTTP/1.1" 200 7525
INFO 2024-02-29 22:12:21,124 basehttp 18000 12112 "GET /admin/myapp_2/client/add/ HTTP/1.1" 200 12140
INFO 2024-02-29 22:12:21,137 basehttp 18000 12112 "GET /admin/jsi18n/ HTTP/1.1" 200 17546
INFO 2024-02-29 22:12:23,301 basehttp 18000 12112 "GET /admin/ HTTP/1.1" 200 7525
INFO 2024-02-29 22:12:24,613 basehttp 18000 12112 "GET /admin/myapp_2/order/add/ HTTP/1.1" 200 15371
INFO 2024-02-29 22:12:24,629 basehttp 18000 12112 "GET /admin/jsi18n/ HTTP/1.1" 200 17546
INFO 2024-02-29 22:12:31,053 basehttp 18000 12112 "GET /admin/myapp_2/client/add/?_to_field=id&_popup=1 HTTP/1.1" 200 7174
INFO 2024-02-29 22:12:31,069 basehttp 18000 12112 "GET /admin/jsi18n/ HTTP/1.1" 200 17546
INFO 2024-02-29 22:12:36,488 basehttp 18000 12112 "GET /admin/ HTTP/1.1" 200 7525
INFO 2024-02-29 22:13:01,392 autoreload 18000 22332 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\shop_project\urls.py changed, reloading.
INFO 2024-02-29 22:13:02,033 autoreload 21204 4716 Watching for file changes with StatReloader
INFO 2024-02-29 22:13:51,581 basehttp 21204 16460 "GET /admin/myapp_2/product/add/ HTTP/1.1" 200 12702
INFO 2024-02-29 22:13:51,599 basehttp 21204 16460 "GET /admin/jsi18n/ HTTP/1.1" 200 17546
INFO 2024-02-29 22:15:14,830 basehttp 21204 16460 "GET /admin/myapp_2/client/ HTTP/1.1" 200 12731
INFO 2024-02-29 22:15:14,843 basehttp 21204 3752 "GET /admin/jsi18n/ HTTP/1.1" 200 17546
INFO 2024-02-29 22:15:14,875 basehttp 21204 16460 "GET /static/admin/css/changelists.css HTTP/1.1" 200 6811
INFO 2024-02-29 22:15:14,893 basehttp 21204 16460 "GET /static/admin/js/filters.js HTTP/1.1" 200 978
INFO 2024-02-29 22:15:14,900 basehttp 21204 16460 "GET /static/admin/img/tooltag-add.svg HTTP/1.1" 200 331
INFO 2024-02-29 22:15:17,715 basehttp 21204 16460 "GET /admin/myapp_2/order/ HTTP/1.1" 200 14332
INFO 2024-02-29 22:15:17,729 basehttp 21204 16460 "GET /admin/jsi18n/ HTTP/1.1" 200 17546
INFO 2024-02-29 22:15:21,048 basehttp 21204 16460 "GET /admin/myapp_2/order/5/change/ HTTP/1.1" 200 16201
INFO 2024-02-29 22:15:21,065 basehttp 21204 16460 "GET /admin/jsi18n/ HTTP/1.1" 200 17546
INFO 2024-02-29 22:15:30,879 basehttp 21204 16460 "GET /admin/myapp_2/order/ HTTP/1.1" 200 14332
INFO 2024-02-29 22:15:30,892 basehttp 21204 16460 "GET /admin/jsi18n/ HTTP/1.1" 200 17546
INFO 2024-02-29 22:15:54,533 basehttp 21204 16460 "GET /admin/myapp_2/client/ HTTP/1.1" 200 12731
INFO 2024-02-29 22:15:54,548 basehttp 21204 16460 "GET /admin/jsi18n/ HTTP/1.1" 200 17546
INFO 2024-02-29 22:15:56,616 basehttp 21204 16460 "GET /admin/myapp_2/product/ HTTP/1.1" 200 13341
INFO 2024-02-29 22:15:56,629 basehttp 21204 16460 "GET /admin/jsi18n/ HTTP/1.1" 200 17546
INFO 2024-02-29 22:18:13,236 autoreload 21204 4716 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\admin.py changed, reloading.
INFO 2024-02-29 22:18:13,687 autoreload 25448 1488 Watching for file changes with StatReloader
INFO 2024-02-29 22:18:25,082 autoreload 25448 1488 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\admin.py changed, reloading.
INFO 2024-02-29 22:18:25,549 autoreload 5256 22084 Watching for file changes with StatReloader
INFO 2024-02-29 22:18:36,923 autoreload 5256 22084 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\admin.py changed, reloading.
INFO 2024-02-29 22:18:37,334 autoreload 6776 16128 Watching for file changes with StatReloader
INFO 2024-02-29 22:18:44,639 autoreload 6776 16128 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\admin.py changed, reloading.
INFO 2024-02-29 22:18:45,033 autoreload 22748 25968 Watching for file changes with StatReloader
INFO 2024-02-29 22:19:04,462 autoreload 22748 25968 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\admin.py changed, reloading.
INFO 2024-02-29 22:19:04,894 autoreload 2952 14400 Watching for file changes with StatReloader
INFO 2024-02-29 22:19:29,515 autoreload 2952 14400 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\admin.py changed, reloading.
INFO 2024-02-29 22:19:29,952 autoreload 21404 25220 Watching for file changes with StatReloader
INFO 2024-02-29 22:19:48,436 autoreload 21404 25220 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\admin.py changed, reloading.
INFO 2024-02-29 22:19:48,915 autoreload 24048 23868 Watching for file changes with StatReloader
INFO 2024-02-29 22:19:49,726 basehttp 24048 25024 "GET /admin/myapp_2/product/ HTTP/1.1" 200 12681
INFO 2024-02-29 22:19:49,745 basehttp 24048 25024 "GET /admin/jsi18n/ HTTP/1.1" 200 17546
INFO 2024-02-29 22:19:55,579 basehttp 24048 25024 "GET /admin/myapp_2/order/ HTTP/1.1" 200 14332
INFO 2024-02-29 22:19:55,593 basehttp 24048 25024 "GET /admin/jsi18n/ HTTP/1.1" 200 17546
INFO 2024-02-29 22:20:03,868 basehttp 24048 25024 "GET /admin/myapp_2/client/ HTTP/1.1" 200 12731
INFO 2024-02-29 22:20:03,886 basehttp 24048 25024 "GET /admin/jsi18n/ HTTP/1.1" 200 17546
INFO 2024-02-29 22:20:05,336 basehttp 24048 25024 "GET /admin/myapp_2/product/ HTTP/1.1" 200 12681
INFO 2024-02-29 22:20:05,352 basehttp 24048 25024 "GET /admin/jsi18n/ HTTP/1.1" 200 17546
INFO 2024-02-29 22:20:11,657 basehttp 24048 25024 "GET /admin/myapp_2/product/3/change/ HTTP/1.1" 200 13612
INFO 2024-02-29 22:20:11,671 basehttp 24048 25024 "GET /admin/jsi18n/ HTTP/1.1" 200 17546
INFO 2024-02-29 22:20:29,750 basehttp 24048 25024 "POST /admin/myapp_2/product/3/change/ HTTP/1.1" 302 0
INFO 2024-02-29 22:20:29,767 basehttp 24048 25024 "GET /admin/myapp_2/product/ HTTP/1.1" 200 13129
INFO 2024-02-29 22:20:29,786 basehttp 24048 25024 "GET /admin/jsi18n/ HTTP/1.1" 200 17546
INFO 2024-02-29 22:20:29,854 basehttp 24048 25024 "GET /static/admin/img/icon-yes.svg HTTP/1.1" 200 436
INFO 2024-02-29 22:20:34,721 basehttp 24048 25024 "GET /admin/myapp_2/product/4/change/ HTTP/1.1" 200 13772
INFO 2024-02-29 22:20:34,735 basehttp 24048 25024 "GET /admin/jsi18n/ HTTP/1.1" 200 17546
INFO 2024-02-29 22:20:45,076 basehttp 24048 25024 "POST /admin/myapp_2/product/4/change/ HTTP/1.1" 302 0
INFO 2024-02-29 22:20:45,096 basehttp 24048 25024 "GET /admin/myapp_2/product/ HTTP/1.1" 200 13199
INFO 2024-02-29 22:20:45,117 basehttp 24048 25024 "GET /admin/jsi18n/ HTTP/1.1" 200 17546
INFO 2024-02-29 22:21:51,536 autoreload 24048 23868 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\admin.py changed, reloading.
INFO 2024-02-29 22:21:51,987 autoreload 18536 2528 Watching for file changes with StatReloader
INFO 2024-02-29 22:21:53,187 basehttp 18536 22392 "GET /admin/myapp_2/order/ HTTP/1.1" 200 14332
INFO 2024-02-29 22:21:53,207 basehttp 18536 22392 "GET /admin/jsi18n/ HTTP/1.1" 200 17546
INFO 2024-02-29 22:21:55,537 basehttp 18536 22392 "GET /admin/myapp_2/order/ HTTP/1.1" 200 14332
INFO 2024-02-29 22:21:55,556 basehttp 18536 22392 "GET /admin/jsi18n/ HTTP/1.1" 200 17546
INFO 2024-02-29 22:22:17,868 autoreload 18536 2528 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\admin.py changed, reloading.
INFO 2024-02-29 22:22:18,332 autoreload 17216 24668 Watching for file changes with StatReloader
INFO 2024-02-29 22:22:18,552 basehttp 17216 24464 "GET /admin/myapp_2/order/ HTTP/1.1" 200 14332
INFO 2024-02-29 22:22:18,569 basehttp 17216 24464 "GET /admin/jsi18n/ HTTP/1.1" 200 17546
INFO 2024-02-29 22:22:20,549 basehttp 17216 24464 "GET /admin/myapp_2/client/ HTTP/1.1" 200 12731
INFO 2024-02-29 22:22:20,563 basehttp 17216 24464 "GET /admin/jsi18n/ HTTP/1.1" 200 17546
INFO 2024-02-29 22:22:21,312 basehttp 17216 24464 "GET /admin/myapp_2/order/ HTTP/1.1" 200 14332
INFO 2024-02-29 22:22:21,325 basehttp 17216 24464 "GET /admin/jsi18n/ HTTP/1.1" 200 17546
INFO 2024-02-29 22:22:22,625 basehttp 17216 24464 "GET /admin/myapp_2/order/6/change/ HTTP/1.1" 200 16224
INFO 2024-02-29 22:22:22,639 basehttp 17216 24464 "GET /admin/jsi18n/ HTTP/1.1" 200 17546
INFO 2024-02-29 22:22:28,350 basehttp 17216 24464 "GET /admin/myapp_2/order/ HTTP/1.1" 200 14332
INFO 2024-02-29 22:22:28,365 basehttp 17216 24464 "GET /admin/jsi18n/ HTTP/1.1" 200 17546
INFO 2024-02-29 22:22:53,351 autoreload 17216 24668 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\admin.py changed, reloading.
INFO 2024-02-29 22:22:53,942 autoreload 25660 25396 Watching for file changes with StatReloader
INFO 2024-02-29 22:23:46,931 autoreload 25660 25396 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\admin.py changed, reloading.
INFO 2024-02-29 22:23:47,496 autoreload 25272 2944 Watching for file changes with StatReloader
INFO 2024-02-29 22:24:28,299 autoreload 25272 2944 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\admin.py changed, reloading.
INFO 2024-02-29 22:24:28,693 autoreload 4312 23052 Watching for file changes with StatReloader
INFO 2024-02-29 22:24:36,989 autoreload 4312 23052 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\admin.py changed, reloading.
INFO 2024-02-29 22:24:37,369 autoreload 10548 14548 Watching for file changes with StatReloader
INFO 2024-02-29 22:24:40,495 basehttp 10548 5036 "GET /admin/myapp_2/order/ HTTP/1.1" 200 13203
INFO 2024-02-29 22:24:40,514 basehttp 10548 5036 "GET /admin/jsi18n/ HTTP/1.1" 200 17546
INFO 2024-02-29 22:24:57,981 autoreload 10548 14548 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\admin.py changed, reloading.
INFO 2024-02-29 22:24:58,407 autoreload 11684 25648 Watching for file changes with StatReloader
INFO 2024-02-29 22:25:13,847 autoreload 11684 25648 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\admin.py changed, reloading.
INFO 2024-02-29 22:25:14,254 autoreload 7440 13012 Watching for file changes with StatReloader
INFO 2024-02-29 22:28:04,825 autoreload 7440 13012 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\models.py changed, reloading.
INFO 2024-02-29 22:28:05,206 autoreload 15008 15788 Watching for file changes with StatReloader
INFO 2024-02-29 22:28:49,059 autoreload 15008 15788 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\models.py changed, reloading.
INFO 2024-02-29 22:28:49,498 autoreload 22232 24100 Watching for file changes with StatReloader
INFO 2024-02-29 22:34:23,728 autoreload 22232 24100 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\admin.py changed, reloading.
INFO 2024-02-29 22:34:24,160 autoreload 8844 22648 Watching for file changes with StatReloader
INFO 2024-02-29 22:34:27,613 basehttp 8844 23732 "GET /admin/myapp_2/order/ HTTP/1.1" 200 15179
INFO 2024-02-29 22:34:27,642 basehttp 8844 23732 "GET /admin/jsi18n/ HTTP/1.1" 200 17546
INFO 2024-02-29 22:34:56,080 autoreload 8844 22648 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\admin.py changed, reloading.
INFO 2024-02-29 22:34:56,529 autoreload 16172 14712 Watching for file changes with StatReloader
INFO 2024-02-29 22:34:58,089 basehttp 16172 25896 "GET /admin/myapp_2/order/ HTTP/1.1" 200 13605
INFO 2024-02-29 22:34:58,109 basehttp 16172 25896 "GET /admin/jsi18n/ HTTP/1.1" 200 17546
INFO 2024-02-29 22:35:01,855 basehttp 16172 25896 "GET /admin/myapp_2/order/6/change/ HTTP/1.1" 200 16224
INFO 2024-02-29 22:35:01,869 basehttp 16172 25896 "GET /admin/jsi18n/ HTTP/1.1" 200 17546
INFO 2024-02-29 22:35:23,986 basehttp 16172 25896 "GET /admin/myapp_2/order/ HTTP/1.1" 200 13605
INFO 2024-02-29 22:46:20,164 autoreload 16172 14712 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\admin.py changed, reloading.
INFO 2024-02-29 22:46:21,023 autoreload 18152 11276 Watching for file changes with StatReloader
INFO 2024-02-29 22:46:21,402 basehttp 18152 15556 "GET /admin/myapp_2/order/ HTTP/1.1" 200 14020
INFO 2024-02-29 22:46:21,431 basehttp 18152 15556 "GET /admin/jsi18n/ HTTP/1.1" 200 17546
INFO 2024-02-29 22:46:44,801 basehttp 18152 15556 "GET /admin/myapp_2/order/5/change/ HTTP/1.1" 200 16224
INFO 2024-02-29 22:46:44,816 basehttp 18152 15556 "GET /admin/jsi18n/ HTTP/1.1" 200 17546
INFO 2024-02-29 22:46:50,740 basehttp 18152 15556 "GET /admin/myapp_2/order/ HTTP/1.1" 200 14020
INFO 2024-02-29 22:46:58,410 basehttp 18152 15556 "POST /admin/myapp_2/order/ HTTP/1.1" 200 9311
INFO 2024-02-29 22:46:58,463 basehttp 18152 15556 "GET /static/admin/js/cancel.js HTTP/1.1" 200 884
INFO 2024-02-29 22:47:02,875 basehttp 18152 15556 "POST /admin/myapp_2/order/ HTTP/1.1" 302 0
INFO 2024-02-29 22:47:02,900 basehttp 18152 15556 "GET /admin/myapp_2/order/ HTTP/1.1" 200 12855
INFO 2024-02-29 22:47:02,919 basehttp 18152 15556 "GET /admin/jsi18n/ HTTP/1.1" 200 17546
INFO 2024-02-29 22:49:16,046 autoreload 18152 11276 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\admin.py changed, reloading.
INFO 2024-02-29 22:49:16,935 autoreload 23556 22724 Watching for file changes with StatReloader
INFO 2024-02-29 22:49:52,724 autoreload 23556 22724 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\admin.py changed, reloading.
INFO 2024-02-29 22:49:53,125 autoreload 24432 16860 Watching for file changes with StatReloader
INFO 2024-02-29 22:49:56,138 basehttp 24432 18652 "GET /admin/myapp_2/order/ HTTP/1.1" 200 12720
INFO 2024-02-29 22:49:56,153 basehttp 24432 18652 "GET /admin/jsi18n/ HTTP/1.1" 200 17546
INFO 2024-02-29 22:50:00,299 basehttp 24432 18652 "GET /admin/myapp_2/order/ HTTP/1.1" 200 12720
INFO 2024-02-29 22:50:00,312 basehttp 24432 18652 "GET /admin/jsi18n/ HTTP/1.1" 200 17546
INFO 2024-02-29 22:50:26,995 autoreload 24432 16860 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\admin.py changed, reloading.
INFO 2024-02-29 22:50:27,463 autoreload 13832 24940 Watching for file changes with StatReloader
ERROR 2024-02-29 22:50:27,794 log 13832 11940 Internal Server Error: /admin/myapp_2/order/
Traceback (most recent call last):
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\models\options.py", line 681, in get_field
    return self.fields_map[field_name]
           ~~~~~~~~~~~~~~~^^^^^^^^^^^^
KeyError: 'get_client_list'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\contrib\admin\utils.py", line 289, in lookup_field
    f = _get_non_gfk_field(opts, name)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\contrib\admin\utils.py", line 322, in _get_non_gfk_field
    field = opts.get_field(name)
            ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\models\options.py", line 683, in get_field
    raise FieldDoesNotExist(
django.core.exceptions.FieldDoesNotExist: Order has no field named 'get_client_list'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\core\handlers\base.py", line 220, in _get_response
    response = response.render()
               ^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\response.py", line 114, in render
    self.content = self.rendered_content
                   ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\response.py", line 92, in rendered_content
    return template.render(context, self._request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\backends\django.py", line 61, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\contrib\admin\templatetags\base.py", line 45, in render
    return super().render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\library.py", line 258, in render
    _dict = self.func(*resolved_args, **resolved_kwargs)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\contrib\admin\templatetags\admin_list.py", line 345, in result_list
    "results": list(results(cl)),
               ^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\contrib\admin\templatetags\admin_list.py", line 321, in results
    yield ResultList(None, items_for_result(cl, res, None))
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\contrib\admin\templatetags\admin_list.py", line 312, in __init__
    super().__init__(*items)
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\contrib\admin\templatetags\admin_list.py", line 217, in items_for_result
    f, attr, value = lookup_field(field_name, result, cl.model_admin)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\contrib\admin\utils.py", line 298, in lookup_field
    value = attr(obj)
            ^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\admin.py", line 17, in get_client_list
    return ", ".join([client.name for client in obj.client.all()])
                                                ^^^^^^^^^^^^^^
AttributeError: 'Client' object has no attribute 'all'
ERROR 2024-02-29 22:50:27,802 basehttp 13832 11940 "GET /admin/myapp_2/order/ HTTP/1.1" 500 391656
INFO 2024-02-29 22:53:08,008 autoreload 13832 24940 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\admin.py changed, reloading.
INFO 2024-02-29 22:53:08,418 autoreload 2820 6776 Watching for file changes with StatReloader
ERROR 2024-02-29 22:53:14,450 log 2820 19604 Internal Server Error: /admin/myapp_2/order/
Traceback (most recent call last):
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\models\options.py", line 681, in get_field
    return self.fields_map[field_name]
           ~~~~~~~~~~~~~~~^^^^^^^^^^^^
KeyError: 'get_client_list'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\contrib\admin\utils.py", line 289, in lookup_field
    f = _get_non_gfk_field(opts, name)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\contrib\admin\utils.py", line 322, in _get_non_gfk_field
    field = opts.get_field(name)
            ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\models\options.py", line 683, in get_field
    raise FieldDoesNotExist(
django.core.exceptions.FieldDoesNotExist: Order has no field named 'get_client_list'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\core\handlers\base.py", line 220, in _get_response
    response = response.render()
               ^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\response.py", line 114, in render
    self.content = self.rendered_content
                   ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\response.py", line 92, in rendered_content
    return template.render(context, self._request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\backends\django.py", line 61, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\contrib\admin\templatetags\base.py", line 45, in render
    return super().render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\library.py", line 258, in render
    _dict = self.func(*resolved_args, **resolved_kwargs)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\contrib\admin\templatetags\admin_list.py", line 345, in result_list
    "results": list(results(cl)),
               ^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\contrib\admin\templatetags\admin_list.py", line 321, in results
    yield ResultList(None, items_for_result(cl, res, None))
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\contrib\admin\templatetags\admin_list.py", line 312, in __init__
    super().__init__(*items)
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\contrib\admin\templatetags\admin_list.py", line 217, in items_for_result
    f, attr, value = lookup_field(field_name, result, cl.model_admin)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\contrib\admin\utils.py", line 298, in lookup_field
    value = attr(obj)
            ^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\admin.py", line 17, in get_client_list
    return ", ".join([client.name for client in obj.client.name])
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\admin.py", line 17, in <listcomp>
    return ", ".join([client.name for client in obj.client.name])
                      ^^^^^^^^^^^
AttributeError: 'str' object has no attribute 'name'
ERROR 2024-02-29 22:53:14,457 basehttp 2820 19604 "GET /admin/myapp_2/order/ HTTP/1.1" 500 395682
ERROR 2024-02-29 22:53:16,947 log 2820 19604 Internal Server Error: /admin/myapp_2/order/
Traceback (most recent call last):
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\models\options.py", line 681, in get_field
    return self.fields_map[field_name]
           ~~~~~~~~~~~~~~~^^^^^^^^^^^^
KeyError: 'get_client_list'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\contrib\admin\utils.py", line 289, in lookup_field
    f = _get_non_gfk_field(opts, name)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\contrib\admin\utils.py", line 322, in _get_non_gfk_field
    field = opts.get_field(name)
            ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\models\options.py", line 683, in get_field
    raise FieldDoesNotExist(
django.core.exceptions.FieldDoesNotExist: Order has no field named 'get_client_list'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\core\handlers\base.py", line 220, in _get_response
    response = response.render()
               ^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\response.py", line 114, in render
    self.content = self.rendered_content
                   ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\response.py", line 92, in rendered_content
    return template.render(context, self._request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\backends\django.py", line 61, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\contrib\admin\templatetags\base.py", line 45, in render
    return super().render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\library.py", line 258, in render
    _dict = self.func(*resolved_args, **resolved_kwargs)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\contrib\admin\templatetags\admin_list.py", line 345, in result_list
    "results": list(results(cl)),
               ^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\contrib\admin\templatetags\admin_list.py", line 321, in results
    yield ResultList(None, items_for_result(cl, res, None))
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\contrib\admin\templatetags\admin_list.py", line 312, in __init__
    super().__init__(*items)
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\contrib\admin\templatetags\admin_list.py", line 217, in items_for_result
    f, attr, value = lookup_field(field_name, result, cl.model_admin)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\contrib\admin\utils.py", line 298, in lookup_field
    value = attr(obj)
            ^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\admin.py", line 17, in get_client_list
    return ", ".join([client.name for client in obj.client.name])
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\admin.py", line 17, in <listcomp>
    return ", ".join([client.name for client in obj.client.name])
                      ^^^^^^^^^^^
AttributeError: 'str' object has no attribute 'name'
ERROR 2024-02-29 22:53:16,956 basehttp 2820 19604 "GET /admin/myapp_2/order/ HTTP/1.1" 500 395682
ERROR 2024-02-29 22:53:17,640 log 2820 19604 Internal Server Error: /admin/myapp_2/order/
Traceback (most recent call last):
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\models\options.py", line 681, in get_field
    return self.fields_map[field_name]
           ~~~~~~~~~~~~~~~^^^^^^^^^^^^
KeyError: 'get_client_list'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\contrib\admin\utils.py", line 289, in lookup_field
    f = _get_non_gfk_field(opts, name)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\contrib\admin\utils.py", line 322, in _get_non_gfk_field
    field = opts.get_field(name)
            ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\models\options.py", line 683, in get_field
    raise FieldDoesNotExist(
django.core.exceptions.FieldDoesNotExist: Order has no field named 'get_client_list'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\core\handlers\base.py", line 220, in _get_response
    response = response.render()
               ^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\response.py", line 114, in render
    self.content = self.rendered_content
                   ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\response.py", line 92, in rendered_content
    return template.render(context, self._request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\backends\django.py", line 61, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\contrib\admin\templatetags\base.py", line 45, in render
    return super().render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\library.py", line 258, in render
    _dict = self.func(*resolved_args, **resolved_kwargs)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\contrib\admin\templatetags\admin_list.py", line 345, in result_list
    "results": list(results(cl)),
               ^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\contrib\admin\templatetags\admin_list.py", line 321, in results
    yield ResultList(None, items_for_result(cl, res, None))
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\contrib\admin\templatetags\admin_list.py", line 312, in __init__
    super().__init__(*items)
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\contrib\admin\templatetags\admin_list.py", line 217, in items_for_result
    f, attr, value = lookup_field(field_name, result, cl.model_admin)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\contrib\admin\utils.py", line 298, in lookup_field
    value = attr(obj)
            ^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\admin.py", line 17, in get_client_list
    return ", ".join([client.name for client in obj.client.name])
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\admin.py", line 17, in <listcomp>
    return ", ".join([client.name for client in obj.client.name])
                      ^^^^^^^^^^^
AttributeError: 'str' object has no attribute 'name'
ERROR 2024-02-29 22:53:17,653 basehttp 2820 19604 "GET /admin/myapp_2/order/ HTTP/1.1" 500 395682
ERROR 2024-02-29 22:53:18,736 log 2820 19604 Internal Server Error: /admin/myapp_2/order/
Traceback (most recent call last):
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\models\options.py", line 681, in get_field
    return self.fields_map[field_name]
           ~~~~~~~~~~~~~~~^^^^^^^^^^^^
KeyError: 'get_client_list'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\contrib\admin\utils.py", line 289, in lookup_field
    f = _get_non_gfk_field(opts, name)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\contrib\admin\utils.py", line 322, in _get_non_gfk_field
    field = opts.get_field(name)
            ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\db\models\options.py", line 683, in get_field
    raise FieldDoesNotExist(
django.core.exceptions.FieldDoesNotExist: Order has no field named 'get_client_list'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\core\handlers\base.py", line 220, in _get_response
    response = response.render()
               ^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\response.py", line 114, in render
    self.content = self.rendered_content
                   ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\response.py", line 92, in rendered_content
    return template.render(context, self._request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\backends\django.py", line 61, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\contrib\admin\templatetags\base.py", line 45, in render
    return super().render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\template\library.py", line 258, in render
    _dict = self.func(*resolved_args, **resolved_kwargs)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\contrib\admin\templatetags\admin_list.py", line 345, in result_list
    "results": list(results(cl)),
               ^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\contrib\admin\templatetags\admin_list.py", line 321, in results
    yield ResultList(None, items_for_result(cl, res, None))
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\contrib\admin\templatetags\admin_list.py", line 312, in __init__
    super().__init__(*items)
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\contrib\admin\templatetags\admin_list.py", line 217, in items_for_result
    f, attr, value = lookup_field(field_name, result, cl.model_admin)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\venv\Lib\site-packages\django\contrib\admin\utils.py", line 298, in lookup_field
    value = attr(obj)
            ^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\admin.py", line 17, in get_client_list
    return ", ".join([client.name for client in obj.client.name])
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\admin.py", line 17, in <listcomp>
    return ", ".join([client.name for client in obj.client.name])
                      ^^^^^^^^^^^
AttributeError: 'str' object has no attribute 'name'
ERROR 2024-02-29 22:53:18,746 basehttp 2820 19604 "GET /admin/myapp_2/order/ HTTP/1.1" 500 395682
INFO 2024-02-29 22:54:43,966 autoreload 2820 6776 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\admin.py changed, reloading.
INFO 2024-02-29 22:54:44,360 autoreload 11888 26148 Watching for file changes with StatReloader
INFO 2024-02-29 22:56:47,836 autoreload 11888 26148 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\admin.py changed, reloading.
INFO 2024-02-29 22:56:48,232 autoreload 16432 9388 Watching for file changes with StatReloader
WARNING 2024-02-29 22:56:50,694 log 16432 24568 Not Found: /
WARNING 2024-02-29 22:56:50,695 basehttp 16432 24568 "GET / HTTP/1.1" 404 2399
INFO 2024-02-29 22:56:54,959 basehttp 16432 24568 "GET /admin/myapp_2/order/ HTTP/1.1" 200 13077
INFO 2024-02-29 22:56:54,980 basehttp 16432 24568 "GET /admin/jsi18n/ HTTP/1.1" 200 17546
INFO 2024-02-29 22:56:58,039 basehttp 16432 24568 "GET /admin/myapp_2/order/6/change/ HTTP/1.1" 200 16224
INFO 2024-02-29 22:56:58,056 basehttp 16432 24568 "GET /admin/jsi18n/ HTTP/1.1" 200 17546
INFO 2024-02-29 22:56:59,580 basehttp 16432 24568 "GET /admin/myapp_2/order/ HTTP/1.1" 200 13077
INFO 2024-02-29 23:19:53,031 basehttp 16432 20152 "GET /admin/myapp_2/order/ HTTP/1.1" 200 13077
INFO 2024-02-29 23:19:53,047 basehttp 16432 20152 "GET /admin/jsi18n/ HTTP/1.1" 200 17546
INFO 2024-02-29 23:19:55,231 basehttp 16432 20152 "GET /admin/myapp_2/product/ HTTP/1.1" 200 12725
INFO 2024-02-29 23:19:55,246 basehttp 16432 20152 "GET /admin/jsi18n/ HTTP/1.1" 200 17546
INFO 2024-02-29 23:19:58,588 basehttp 16432 20152 "GET /admin/myapp_2/order/ HTTP/1.1" 200 13077
INFO 2024-02-29 23:19:58,608 basehttp 16432 20152 "GET /admin/jsi18n/ HTTP/1.1" 200 17546
INFO 2024-02-29 23:19:59,254 basehttp 16432 20152 "GET /admin/myapp_2/client/ HTTP/1.1" 200 12731
INFO 2024-02-29 23:19:59,271 basehttp 16432 20152 "GET /admin/jsi18n/ HTTP/1.1" 200 17546
INFO 2024-02-29 23:22:24,766 autoreload 16432 9388 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\admin.py changed, reloading.
INFO 2024-02-29 23:22:25,610 autoreload 24336 7128 Watching for file changes with StatReloader
INFO 2024-02-29 23:22:26,308 basehttp 24336 22376 "GET /admin/myapp_2/order/ HTTP/1.1" 200 13077
INFO 2024-02-29 23:22:26,334 basehttp 24336 22376 "GET /admin/jsi18n/ HTTP/1.1" 200 17546
INFO 2024-02-29 23:22:27,372 basehttp 24336 22376 "GET /admin/myapp_2/client/ HTTP/1.1" 200 12731
INFO 2024-02-29 23:22:27,388 basehttp 24336 22376 "GET /admin/jsi18n/ HTTP/1.1" 200 17546
INFO 2024-02-29 23:23:02,454 autoreload 24336 7128 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\admin.py changed, reloading.
INFO 2024-02-29 23:23:02,865 autoreload 26176 16428 Watching for file changes with StatReloader
INFO 2024-02-29 23:23:11,202 autoreload 26176 16428 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\myapp_2\admin.py changed, reloading.
INFO 2024-02-29 23:23:11,646 autoreload 22440 16996 Watching for file changes with StatReloader
INFO 2024-02-29 23:23:12,321 basehttp 22440 21368 "GET /admin/myapp_2/client/ HTTP/1.1" 200 12880
INFO 2024-02-29 23:23:12,338 basehttp 22440 21368 "GET /admin/jsi18n/ HTTP/1.1" 200 17546
INFO 2024-02-29 23:23:28,633 basehttp 22440 21368 "GET /admin/myapp_2/client/5/change/ HTTP/1.1" 200 12954
INFO 2024-02-29 23:23:28,653 basehttp 22440 21368 "GET /admin/jsi18n/ HTTP/1.1" 200 17546
INFO 2024-02-29 23:23:34,956 basehttp 22440 21368 "GET /admin/myapp_2/order/ HTTP/1.1" 200 13077
INFO 2024-02-29 23:23:34,973 basehttp 22440 21368 "GET /admin/jsi18n/ HTTP/1.1" 200 17546
INFO 2024-02-29 23:23:37,278 basehttp 22440 21368 "GET /admin/myapp_2/client/ HTTP/1.1" 200 12880
INFO 2024-02-29 23:23:37,299 basehttp 22440 21368 "GET /admin/jsi18n/ HTTP/1.1" 200 17546
INFO 2024-02-29 23:23:41,174 basehttp 22440 21368 "GET /admin/myapp_2/client/4/change/ HTTP/1.1" 200 12849
INFO 2024-02-29 23:23:41,187 basehttp 22440 21368 "GET /admin/jsi18n/ HTTP/1.1" 200 17546
INFO 2024-02-29 23:23:44,621 basehttp 22440 21368 "GET /admin/myapp_2/client/ HTTP/1.1" 200 12880
INFO 2024-02-29 23:23:44,640 basehttp 22440 21368 "GET /admin/jsi18n/ HTTP/1.1" 200 17546
INFO 2024-02-29 23:23:45,981 basehttp 22440 21368 "GET /admin/myapp_2/client/1/change/ HTTP/1.1" 200 12934
INFO 2024-02-29 23:23:45,997 basehttp 22440 21368 "GET /admin/jsi18n/ HTTP/1.1" 200 17546
INFO 2024-02-29 23:23:53,573 basehttp 22440 21368 "GET /admin/myapp_2/client/ HTTP/1.1" 200 12880
INFO 2024-02-29 23:23:53,594 basehttp 22440 21368 "GET /admin/jsi18n/ HTTP/1.1" 200 17546
INFO 2024-02-29 23:26:05,210 basehttp 22440 21368 "GET /admin/myapp_2/product/ HTTP/1.1" 200 12725
INFO 2024-02-29 23:26:05,222 basehttp 22440 21368 "GET /admin/jsi18n/ HTTP/1.1" 200 17546
INFO 2024-02-29 23:26:06,483 basehttp 22440 21368 "GET /admin/myapp_2/product/5/change/ HTTP/1.1" 200 13694
INFO 2024-02-29 23:26:06,502 basehttp 22440 21368 "GET /admin/jsi18n/ HTTP/1.1" 200 17546
INFO 2024-03-07 21:30:50,912 autoreload 24700 22836 Watching for file changes with StatReloader
INFO 2024-03-07 22:23:28,166 autoreload 10540 8964 Watching for file changes with StatReloader
INFO 2024-03-07 22:27:42,677 autoreload 10540 8964 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\shop_project\settings.py changed, reloading.
INFO 2024-03-07 22:27:43,187 autoreload 19140 19340 Watching for file changes with StatReloader
INFO 2024-03-07 22:27:50,525 autoreload 19140 19340 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\shop_project\settings.py changed, reloading.
INFO 2024-03-07 22:27:50,947 autoreload 23408 14056 Watching for file changes with StatReloader
INFO 2024-03-07 22:27:57,225 autoreload 23408 14056 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\shop_project\settings.py changed, reloading.
INFO 2024-03-07 22:27:57,633 autoreload 18436 4548 Watching for file changes with StatReloader
INFO 2024-03-07 22:28:04,947 autoreload 18436 4548 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\shop_project\settings.py changed, reloading.
INFO 2024-03-07 22:28:05,334 autoreload 16212 25028 Watching for file changes with StatReloader
INFO 2024-03-07 22:30:23,717 autoreload 16212 25028 C:\Users\Никита\Desktop\DJANGO\ShopProject\Shop_project_HW\shop_project\shop_project\settings.py changed, reloading.
INFO 2024-03-07 22:30:24,168 autoreload 9764 18820 Watching for file changes with StatReloader
INFO 2024-03-07 22:34:32,330 autoreload 17992 23784 Watching for file changes with StatReloader
